reactive-banana/src/Reactive/Banana/Prim/Evaluation.hs

```diff
@@ -107,14 +108,14 @@ insertNodes :: RWS.Tuple BuildR (EvalPW, BuildW) Lazy.Vault -> [SomeNode] -> Que
 insertNodes (RWS.Tuple (time,_) _ _) = go
     where
     go :: [SomeNode] -> Queue SomeNode -> IO (Queue SomeNode)
-    go []              q = return q
-    go (node@(P p):xs) q = do
+    go []              !q = return q
```

Comment:
This is a *big* reduction in allocations, perhaps the biggest in this branch. I think we could replace `insertNodes` with `foldl'` or `foldr` and this might be a bit more obvious.